libraries/matrix/api/src/main/kotlin/io/element/android/libraries/matrix/api/permalink/PermalinkData.kt

```diff
@@ -27,28 +29,44 @@ import kotlinx.collections.immutable.ImmutableList
  */
 @Immutable
 sealed interface PermalinkData {
-    data class RoomLink(
-        val roomIdOrAlias: String,
-        val isRoomAlias: Boolean,
-        val eventId: String?,
+    sealed interface RoomLink : PermalinkData {
         val viaParameters: ImmutableList<String>
-    ) : PermalinkData {
-        fun getRoomId(): RoomId? {
-            return roomIdOrAlias.takeIf { !isRoomAlias }?.let(::RoomId)
-        }
+    }
+
+    data class RoomIdLink(
+        val roomId: RoomId,
+        override val viaParameters: ImmutableList<String>
+    ) : RoomLink
+
+    data class RoomAliasLink(
+        val roomAlias: String,
+        override val viaParameters: ImmutableList<String>
+    ) : RoomLink
 
-        fun getRoomAlias(): String? {
-            return roomIdOrAlias.takeIf { isRoomAlias }
-        }
+    sealed interface EventLink : PermalinkData {
+        val eventId: EventId
+        val viaParameters: ImmutableList<String>
     }
 
+    data class EventIdLink(
+        val roomId: RoomId,
+        override val eventId: EventId,
+        override val viaParameters: ImmutableList<String>
+    ) : EventLink
+
+    data class EventIdAliasLink(
+        val roomAlias: String,
```

Comment:
create `RoomAlias` and `RoomOrAliasId` value classes? I really think it'll be easier to manage like that

Reply:
Yes, maybe, but I thought you created it?

And I am pretty sure that when we build a permalink for an event, the roomId is used, even if a room Alias is availalbe. The main reason is that roomId are immutable (module room upgrade...), and aliases can change.

Can this be handled separately?